remove redundant dynamic_dim

avikchaudhuri opened the pull request:
Differential Revision: D48618472



cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov

angelayi commented on torch/fx/experimental/symbolic_shapes.py:
```diff
@@ -1959,13 +1980,13 @@ def print_results(grouped, indent, result_fn):
             print_results(
                 grouped_dynamic_results,
                 indent * 2,
-                lambda result: f"{remove_default_lower_bound(result)},",
```
Should `remove_default_lower_bound` be deleted then?

gmagogsfm replied:
OOC, why is `remove_default_lower_bound` not called any more?

avikchaudhuri replied:
Moved its functionality upstream.

gmagogsfm commented on test/dynamo/test_export.py:
```diff
@@ -2264,6 +2264,22 @@ def my_dyn_fn(a, b, c):
+        a = torch.randn(3)
+        b = torch.randn(3)
+        with self.assertRaisesRegex(
+            torch._dynamo.exc.UserError, "\\[\n.*\n.*dynamic_dim.*==.*dynamic_dim.*\n.*\\]"
```
Optional nit:

This regex is overly specific, and it can become a "change detector". A better way is to detect that there is no patterns like "dynamic_dim(t, idx)," on its own line.

gmagogsfm commented on torch/fx/experimental/symbolic_shapes.py:
```diff
@@ -1885,6 +1888,24 @@ def forced_specializations(self):
             if s in self._marked_dynamic
         ])
 
+    def remove_redundant_dynamic_results(self):
```
a bit of comment explaining its behavior would be nice.

avikchaudhuri replied:
yeah, added

avikchaudhuri commented:
@pytorchbot merge

avikchaudhuri commented:
@pytorchbot merge

avikchaudhuri committed:
remove redundant dynamic_dim (#107815)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/107815

Test Plan: added test

Reviewed By: tugsbayasgalan

Differential Revision: D48618472

fbshipit-source-id: 0078811f405fd8ad5827edffcdcd3943091db62b

avikchaudhuri commented:
@pytorchbot merge
